### PR TITLE
[containerapp] Fix `az containerapp update --secret-volume-mount` appending to NoneType object bug

### PR DESCRIPTION
**Pull request by smzeng**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

az containerapp update

fixes bug for using the update command to mount a secret volume, checks for if volume is null since volume is now a NoneType object and mounts volumeMount as an array 

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Review by Juancpani — APPROVED** [review]

**Review by pagariyaalok — APPROVED** [review]
:shipit:

**Comment by smzeng** [comment]
@yonzhan this has been reviewed by the ACA team and ready for review from the CLI team

File: src/containerapp/HISTORY.rst [hunk 1]
```diff
@@ -7,6 +7,7 @@ upcoming
 0.3.41
 ++++++
 * 'az containerapp up/create': enable support for no Dockerfile cases with --repo
+* 'az containerapp update': fix bug for mounting secret volumes using --secret-volume-mount
```
Comment (Greedygre) [inline src/containerapp/HISTORY.rst:10]: please add description in upcoming section.

**Comment by Greedygre** [comment]
Recommend PR title:

`az containerapp update`: `--secret-volume-mount` fix appending to NoneType object bug

**Comment by Greedygre** [comment]
As we have the same command and feature in Azure-cli, please fix in https://github.com/Azure/azure-cli/tree/dev/src/azure-cli/azure/cli/command_modules/containerapp too.

File: src/containerapp/HISTORY.rst (hunk 1 above)
Comment (Juancpani) [reply src/containerapp/HISTORY.rst]: Done!

**Comment by Juancpani** [comment]
> Recommend PR title:
> 
> `az containerapp update`: `--secret-volume-mount` fix appending to NoneType object bug

I sadly don't have the ability to change the title and Stacy is OOF, next time though.

**Review by Greedygre — APPROVED** [review]

**Comment by Greedygre** [comment]
Hi @zhoxing-ms 
Could you please help to review this PR? Thanks.

File: src/containerapp/azext_containerapp/tests/latest/test_containerapp_mount_secret_volume.py [hunk 2]
```diff
@@ -38,5 +38,50 @@ def test_container_app_mount_secret_e2e(self, resource_group):
...
+        self.cmd(f'az containerapp update -n {app} -g {resource_group} --secret-volume-mount "mnt/newpath"')    
+        
+        self.cmd('containerapp show -g {} -n {}'.format(resource_group, app), checks=[
+            JMESPathCheck('properties.template.containers[0].volumeMounts[0].mountPath', 'mnt/newpath'), 
+        ])
+    
+    @AllowLargeResponse(8192)
+    @ResourceGroupPreparer(location="northcentralus")
+    def test_container_app_mount_secret_update_e2e(self, resource_group):
+        import requests
```
Comment (wangzelin007) [inline src/containerapp/azext_containerapp/tests/latest/test_containerapp_mount_secret_volume.py:54]: Why is the import requests needed here?

**Review by wangzelin007 — APPROVED** [review]

File: src/containerapp/azext_containerapp/tests/latest/test_containerapp_mount_secret_volume.py (hunk 2 above)
Comment (Greedygre) [reply src/containerapp/azext_containerapp/tests/latest/test_containerapp_mount_secret_volume.py:54]: I will remove unused import in PR: https://github.com/Azure/azure-cli-extensions/pull/6807

**Pull request merged**